Mathlib/Tactic/Repeat.lean

```diff
@@ -0,0 +1,39 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import Lean
+import Mathlib.Tactic.Core
+
+/-!
+# Repeating tactics
+
+We provide a variation on `repeat`:
+* `repeat1 tac` behaves like `repeat`, but fails unless `tac` succeeds at least once.
+
+Note also the macro
+* `repeat' tac` defined in `Std`, which runs `tac` on each subgoal, continuing after failures.
+
+We provide an analogue of `repeat'` as a combinator in `MetaM`.
+-/
+
+/--
+`repeat1 tac` applies `tac` to main goal at least once. If the application succeeds,
+the tactic is applied recursively to the generated subgoals until it eventually fails.
```

Comment:
This description isn't correct, it only applies `tac` to the main goal, and then the first generated subgoal, and so on until it fails. Did you mean to use `<;>` instead of `;`?

Reply:
I only moved this one from `Mathlib/Tactic/Core.lean`, where it was introduced in #326.

How about we leave the definition as is (with `;` rather than `<;>`), and change the docstring to

```
`repeat1 tac` repeatedly applies `tac` until it fails. `repeat1` succeeds if and only if `tac` succeeds at least once.
```